hazelcast/src/main/java/com/hazelcast/internal/networking/AbstractChannel.java

```diff
@@ -108,27 +108,35 @@ public SocketAddress localSocketAddress() {
 
     @Override
     public void connect(InetSocketAddress address, int timeoutMillis) throws IOException {
-        checkNotNull(address, "address");
-        checkNotNegative(timeoutMillis, "timeoutMillis can't be negative");
+        try {
+            checkNotNull(address, "address");
+            checkNotNegative(timeoutMillis, "timeoutMillis can't be negative");
 
-        // since the connect method is blocking, we need to configure blocking.
-        socketChannel.configureBlocking(true);
+            // since the connect method is blocking, we need to configure blocking.
+            socketChannel.configureBlocking(true);
 
-        try {
-            if (timeoutMillis > 0) {
-                socketChannel.socket().connect(address, timeoutMillis);
-            } else {
-                socketChannel.connect(address);
+            try {
+                if (timeoutMillis > 0) {
+                    socketChannel.socket().connect(address, timeoutMillis);
+                } else {
+                    socketChannel.connect(address);
+                }
+            } catch (SocketException ex) {
+                //we want to include the address in the exception.
+                SocketException newEx = new SocketException(ex.getMessage() + " to address " + address);
+                newEx.setStackTrace(ex.getStackTrace());
+                throw newEx;
             }
-        } catch (SocketException ex) {
-            //we want to include the address in the exception.
-            SocketException newEx = new SocketException(ex.getMessage() + " to address " + address);
-            newEx.setStackTrace(ex.getStackTrace());
-            throw newEx;
-        }
 
-        if (logger.isFinestEnabled()) {
-            logger.finest("Successfully connected to: " + address + " using socket " + socketChannel.socket());
+            if (logger.isFinestEnabled()) {
+                logger.finest("Successfully connected to: " + address + " using socket " + socketChannel.socket());
+            }
+        } catch (RuntimeException e) {
+            close();
```

Comment:
shouldn't you catch all exception throws while closing? now you can lose the original exception if there is another exception while closing the channel.

Reply:
Yes, let me change this part `catch(Exception e)`

Reply:
So the thing is here, `connect` throws only IOException. `try` part can not throw any other unchecked exception than `IOException`. Rest will be `RuntimeException` . If I am not mistaken we will not lose any exception in current state. So , current implementations seems OK to me ? @jerrinot wdyt ?